samples/SampleApp/SampleApp.csproj

```diff
@@ -17,8 +15,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="$(AspNetCoreVersion)" />
-    <PackageReference Include="Microsoft.NETCore.Compilers" Version="$(MicrosoftNETCoreCompilersVersion)" PrivateAssets="All" />
```

Comment:
@pakrym built something close to this. But we haven't turned it on by default because it doesn't catch everything. i.e. some packages, like Microsoft.NETCore.Compilers, contribute build-time only assets to the build which don't show up in the final artifacts.

https://github.com/aspnet/BuildTools/blob/da73e105376f5b980593c1519e58d8bcde8fdb65/src/Internal.AspNetCore.Sdk/build/FindUnusedReferences.targets#L6